node/tests/test-PipeTransport.js

```diff
@@ -879,18 +894,60 @@ test('router.pipeToRouter() called twice generates a single PipeTransport pair',
 	// There should be 1 Transport in routerB:
 	// - PipeTransport between routerA and routerB.
 	expect(dump.transportIds.length).toBe(1);
+
+	routerA.close();
+	routerB.close();
 }, 2000);
 
-test('router.createPipeTransport() with fixed port succeeds', async () =>
+test('router.pipeToRouter() called in two Routers passing one to each other as argument generates a single a single PipeTransport pair', async () =>
 {
-	const port = await pickPort({ ip: '127.0.0.1', reserveTimeout: 0 });
-	const pipeTransport = await router1.createPipeTransport(
-		{
-			listenIp : '127.0.0.1',
-			port
-		});
+	// We must close other Routers so previously generated PipeTransports are
+	// closed and removed from the map.
+	router1.close();
+	router2.close();
 
-	expect(pipeTransport.tuple.localPort).toEqual(port);
+	const routerA = await worker.createRouter({ mediaCodecs });
+	const routerB = await worker.createRouter({ mediaCodecs });
+	const transportA = await routerA.createWebRtcTransport({ listenIps: [ '127.0.0.1' ] });
+	const transportB = await routerB.createWebRtcTransport({ listenIps: [ '127.0.0.1' ] });
+	const audioProducerA = await transportA.produce(audioProducerParameters);
+	const audioProducerB = await transportB.produce(audioProducerParameters);
 
-	pipeTransport.close();
+	await Promise.all(
+		[
+			routerA.pipeToRouter(
+				{
+					producerId : audioProducerA.id,
+					router     : routerB
+				}),
+			routerB.pipeToRouter(
+				{
+					producerId : audioProducerB.id,
+					router     : routerA
+				})
+		]);
+
+	// There should be a single PlainTransport pair in both Routers and they must
+	// be the same one
+
+	const key = Router.getPipeTransportPairKey(routerA, routerB);
+	const mapA = routerA.mapRouterPairPipeTransportPairPromiseForTesting;
+	const mapB = routerB.mapRouterPairPipeTransportPairPromiseForTesting;
```

Comment:
This private API can be replaced with an observer that tracks how many transports were created on each router.

Reply:
That's not enough for testing. I need to verify not just PipeTransports created in each Router but also the associated remote PipeTransport of each local PipeTransport.

Reply:
Should be possible to compare by ID. Besides, when only one is created on each side, it is somewhat obvious that they correspond to each other (assuming other tests also pass).

Reply:
Right!

Reply:
Pushed